spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/configclient/ZookeeperConfigServerBootstrapper.java

```diff
@@ -131,67 +132,9 @@ public void initialize(BootstrapRegistry registry) {
-	 *
-	 * We will create many of the same beans in this function as we do above in the initializer above.  We do both
-	 * to maintain compatibility since we are promoting those beans to the main application context.
-	 */
-	static final class ZookeeperFunction implements ConfigServerInstanceProvider.Function {
-
-		private final BootstrapContext context;
-
-		private ZookeeperFunction(BootstrapContext context) {
-			this.context = context;
-		}
-
-		static ZookeeperFunction create(BootstrapContext context) {
-			return new ZookeeperFunction(context);
-		}
-
-		@Override
-		public List<ServiceInstance> apply(String serviceId) {
-			return apply(serviceId, null, null, null);
-		}
-
-		@Override
-		public List<ServiceInstance> apply(String serviceId, Binder binder, BindHandler bindHandler, Log log) {
-			if (binder == null || !isEnabled(binder)) {
-				return Collections.emptyList();
-			}
 
-			ZookeeperProperties properties = context.getOrElse(ZookeeperProperties.class, binder.bind(ZookeeperProperties.PREFIX, Bindable.of(ZookeeperProperties.class))
-					.orElse(new ZookeeperProperties()));
-			RetryPolicy retryPolicy = context.getOrElse(RetryPolicy.class, new ExponentialBackoffRetry(properties.getBaseSleepTimeMs(), properties.getMaxRetries(),
-					properties.getMaxSleepMs()));
-			try {
-				CuratorFramework curatorFramework = context.getOrElse(CuratorFramework.class, CuratorFactory.curatorFramework(properties, retryPolicy, Stream::of,
-						() -> null, () -> null));
-				InstanceSerializer<ZookeeperInstance> serializer = context.getOrElse(InstanceSerializer.class, new JsonInstanceSerializer<>(ZookeeperInstance.class));
-				ZookeeperDiscoveryProperties discoveryProperties = context.getOrElse(ZookeeperDiscoveryProperties.class, binder.bind(ZookeeperDiscoveryProperties.PREFIX, Bindable
-								.of(ZookeeperDiscoveryProperties.class), bindHandler)
-						.orElseGet(() -> new ZookeeperDiscoveryProperties(new InetUtils(new InetUtilsProperties()))));
-				ServiceDiscoveryCustomizer customizer = context.getOrElse(ServiceDiscoveryCustomizer.class, new DefaultServiceDiscoveryCustomizer(curatorFramework, discoveryProperties, serializer));
-				ServiceDiscovery<ZookeeperInstance> serviceDiscovery = customizer.customize(ServiceDiscoveryBuilder.builder(ZookeeperInstance.class));
-				ZookeeperDependencies dependencies = context.getOrElse(ZookeeperDependencies.class, binder.bind(ZookeeperDependencies.PREFIX, Bindable
-								.of(ZookeeperDependencies.class), bindHandler)
-						.orElseGet(ZookeeperDependencies::new));
-				return new ZookeeperDiscoveryClient(serviceDiscovery, dependencies, discoveryProperties).getInstances(serviceId);
-			}
-			catch (Exception e) {
-				log.warn("Error fetching config server instance from Zookeeper", e);
-				return Collections.emptyList();
-			}
-
-		}
+	private static BindHandler getBindHandler(org.springframework.boot.BootstrapContext context) {
+		return context.getOrElse(BindHandler.class, null);
 	}
 
```

Comment:
```suggestion
```
This could probably be inlined & removed now since it's only used in one place (`getPropertyResolver`).